agenta-web/src/components/pages/settings/Secrets/Secrets.tsx

```diff
@@ -41,38 +48,52 @@ export default function Secrets() {
             </Text>
 
             <div className={classes.container}>
-                <Title level={5}>Providers</Title>
+                <Title level={5}>Available Providers</Title>
 
                 <div className={classes.apiContainer}>
-                    <Space direction="horizontal">
-                        <Input.Password
-                            data-cy="openai-api-input"
-                            value={openAiKey}
-                            onChange={(e) => setOpenAiKey(e.target.value)}
-                            addonBefore="OpenAI API Key"
-                            visibilityToggle={false}
-                            className={classes.input}
-                        />
-                        <Button
-                            data-cy="openai-api-save"
-                            disabled={saveDisabled}
-                            onClick={() => {
-                                saveOpenAIKey(openAiKey)
-                                messageAPI.success("The secret is saved")
-                            }}
-                        >
-                            Save
-                        </Button>
-                        <Button
-                            onClick={() => {
-                                removeOpenAIKey()
-                                setOpenAiKey("")
-                                messageAPI.warning("The secret is deleted")
-                            }}
-                        >
-                            Delete
-                        </Button>
-                    </Space>
+                    {llmProviderKeys.map(({title, key}: LlmProvider, i: number) => (
+                        <Space direction="horizontal" key={i}>
+                            <Input.Password
+                                data-cy="openai-api-input"
+                                value={key}
+                                onChange={(e) => {
+                                    const newLlmProviderKeys = [...llmProviderKeys]
+                                    newLlmProviderKeys[i].key = e.target.value
+                                    setLlmProviderKeys(newLlmProviderKeys)
+                                }}
+                                addonBefore={`${title}`}
+                                visibilityToggle={false}
+                                className={classes.input}
+                            />
+                            <Button
+                                data-cy="openai-api-save"
+                                disabled={
+                                    llmProviderKeys[i].key ===
+                                        getLlmProviderKey(llmProviderKeys[i].key) ||
+                                    !llmProviderKeys[i].key
```

Comment:
Can you please replace `llmProviderKeys[i].key` with `key` on all 3 occasions here to keep things concise?

Reply:
Nice, will do!